Core/NWNXCore.cpp

```diff
@@ -145,6 +146,20 @@ NWNXCore::~NWNXCore()
     Shutdown();
 }
 
+void NWNXCore::CleanupPreload()
+{
+    const auto* preload = std::getenv("LD_PRELOAD");
+    const std::regex regex("(([^: ]+)?NWNX_[^: ]+)");
+
+    std::string newPreload = preload;
+    while (std::regex_search(newPreload, regex))
+    {
+        newPreload = std::regex_replace(newPreload, regex, "");
+    }
+
+    setenv("LD_PRELOAD", newPreload.c_str(), true);
+}
+
```

Comment:
I think, that such global changes in user-supplied environment requires some comment about reasons, because this is a very non-standard actions

Reply:
Hi, you're right, sorry, it's easy to forget about context when you've been freshly discussing it elsewhere. For the record, the issue is that sub-processes launched by any nwnx plugin will inherit LD_PRELOAD and attempt to preload NWNX_Core.so. This only works when preloaded onto nwn (nwmain or nwserver), and other process startups will fail. 
@jhett12321 can you please add a comment here as well? Maybe something in the readme too if you find a good place for it.